File: app/src/main/java/jaipurbus/jaipur/tourism/jaipurbus/ui/gallery/adapter/ImagesAdapter.java
package jaipurbus.jaipur.tourism.jaipurbus.ui.gallery.adapter;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

 
import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.codersworld.jplibs.customviews.TouchImageView;
import com.codersworld.jplibs.utils.CommonMethods;

import java.util.ArrayList;

import jaipurbus.jaipur.tourism.jaipurbus.R;

public class ImagesAdapter extends PagerAdapter {
    /* access modifiers changed from: private */
    public Activity _activity;
    private ArrayList<String> _imagePaths = new ArrayList<>();
    private LayoutInflater inflater;

    public ImagesAdapter(Activity activity, ArrayList<String> imagePaths) {
        this._activity = activity;
        this._imagePaths = imagePaths;
    }

    public int getCount() {
        return this._imagePaths.size();
    }
    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == ((RelativeLayout) object);
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        LayoutInflater layoutInflater = (LayoutInflater) this._activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        this.inflater = layoutInflater;
        View viewLayout = layoutInflater.inflate(R.layout.layout_fullscreen_image, container, false);
        CommonMethods.setBackground(_activity,_imagePaths.get(position), viewLayout.findViewById(R.id.rlContainer));

        Glide.with(this._activity).load(this._imagePaths.get(position)).apply(new RequestOptions().placeholder(android.R.drawable.ic_menu_gallery).error(android.R.drawable.ic_menu_gallery)).into((ImageView) (TouchImageView) viewLayout.findViewById(R.id.imgDisplay));
        ((ViewPager) container).addView(viewLayout);
        return viewLayout;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        ((ViewPager) container).removeView((RelativeLayout) object);
    }
}

File: app/src/main/java/jaipurbus/jaipur/tourism/jaipurbus/ui/places/adapters/ImagesAdapter.kt
package jaipurbus.jaipur.tourism.jaipurbus.ui.places.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView 
import com.bumptech.glide.Glide
import com.codersworld.jplibs.loopingviewpager.LoopingPagerAdapter
import com.codersworld.jplibs.utils.CommonMethods
import jaipurbus.jaipur.tourism.jaipurbus.R
import jaipurbus.jaipur.tourism.jaipurbus.ui.gallery.GalleryActivity
import kotlin.collections.ArrayList

class ImagesAdapter(
    var mContext: Context,
    itemList: ArrayList<String>,
    isInfinite: Boolean
) : LoopingPagerAdapter<String>(itemList, isInfinite) {

    override fun getItemViewType(listPosition: Int): Int {
        return VIEW_TYPE_NORMAL
    }

    override fun inflateView(viewType: Int, container: ViewGroup, listPosition: Int): View {
        return LayoutInflater.from(container.context)
            .inflate(R.layout.item_images, container, false)
    }

    override fun bindView(convertView: View, listPosition: Int, viewType: Int) {
        val rlContainer = convertView.findViewById<TextView>(R.id.rlContainer) as RelativeLayout
        val imgIcon = convertView.findViewById<TextView>(R.id.imgBanner) as ImageView
        CommonMethods.setBackground(mContext, itemList!![listPosition], rlContainer)
        Glide.with(mContext)
            .load(itemList!![listPosition])
            .placeholder(R.drawable.app_icon)
            .error(R.drawable.app_icon)
            .into(imgIcon)
        imgIcon.setOnClickListener {
            var mImages: ArrayList<String> = ArrayList()
            for (i in itemList!!.indices) {
                var mBn = itemList!![i]
                mImages.add(mBn)
            }
            if (CommonMethods.isValidArrayList(mImages)) {
                mContext.startActivity(Intent(mContext, GalleryActivity::class.java).putExtra("mData",mImages).putExtra("position", listPosition))
            }
        }
    }

    companion object {
        private const val VIEW_TYPE_NORMAL = 100
        private const val VIEW_TYPE_SPECIAL = 101
    }
}